### ISSUE TITLE
Postgres `RETURNING` not parsing properly

### ISSUE THREAD

**Issue by hiporox** [sqlfluff/sqlfluff#6697]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This is almost identical to the closed issue #5865. The query:
```psql
INSERT INTO prompt_variants (
    test,
    test2
)
SELECT
    test,
    test2
RETURNING 
  test,
  test2;
```
returns the error
``` 
Line 8, Position 1: Found unparsable section: 'RETURNING 
  test,
  test2'
```

### Expected Behaviour

This should not cause any errors as it is valid SQL in Postgres

### Observed Behaviour

The parser fails to understand `RETURNING` statements

### How to reproduce

Using the [Online SQL Fluff](https://online.sqlfluff.com/fluffed?sql=H4sIAJfPwGcC__P0C3YNClHw9AvxVygoys8tKIkvSyzKTMwrKVbQ4OVSAIKS1OISHQTTiJdLk5cr2NXH1TkEl4Ig15DQID9PP3cFkDBcHixrzcsFAAlhETZ1AAAA&dialect=PostgreSQL), you can see that this currently fails (as of 2025-02-27T20:50Z)

You can create a SQL file named `test.sql` with the following contents:
```sql
INSERT INTO prompt_variants (
    test,
    test2
)
SELECT
    test,
    test2
RETURNING 
  test,
  test2;
```

Using the CLI over this file with `sqlfluff --dialect postgres test.sql` will produce the error

### Dialect

postgres

### Version

3.3.1

### Configuration

The configuration is empty, dialect is set with the CLI option

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

I am happy to submit a PR to fix, but will need some guidance on what needs to be done

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by hiporox** [comment]
One thing to note is that it is specifically the combination of `SELECT` and `RETURNING`. Using a `VALUES` instead of a `SELECT` parses properly

**Comment by hiporox** [comment]
Update: It appears that SQLFluff handles the case of `INSERT INTO ... SELECT FROM ... RETURNING`, but doesn't handle the postgres syntax that allows skipping the `FROM` clause when inserting values when using a `unnest` function
